commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/AbstractGenerator.java

```diff
@@ -3,34 +3,39 @@
 import com.squareup.javapoet.JavaFile;
 import com.squareup.javapoet.TypeSpec;
 
+import javax.annotation.processing.Messager;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.util.Elements;
 import javax.lang.model.util.Types;
 
 /**
  * Abstract base class for implementing javapoet based generators - to generate a single class.
+ * The generic attribute T should be used to specify whether the annotation is used on the class level in which {@code T} would be {@link TypeElement} or method level,
+ * in that case T would be {@link ExecutableElement}
  */
-abstract class AbstractGenerator extends BaseAbstractGenerator {
+abstract class AbstractGenerator<T extends Element> extends BaseAbstractGenerator {
 
-    AbstractGenerator(final Elements elements, final Types types) {
-        super(elements, types);
+    AbstractGenerator(final Elements elements, final Types types, Messager messager) {
```

Comment:
Great that you added it here :1st_place_medal:  Just a minor thing: I would like to add `final` the the `messager` parameter here.